cvmfs/catalog.cc

```diff
@@ -408,6 +414,44 @@ void Catalog::DropDatabaseFileOwnership() {
 }
 
 
+Catalog::ExternalDataStatus Catalog::GetExternalDataLocked() const {
+  ExternalDataStatus status = static_cast<ExternalDataStatus>(atomic_read32(&external_data_status_));
+  if (status == Unknown) {
+    bool attr = false;
+    if (database().HasProperty("external_data")) {
+      attr = database().GetProperty<int>("external_data");
+      status = attr ? Present : None;
+    } else {
+      status = parent_ ? parent_->GetExternalDataUnlocked() : Unspecified;
+    }
+    atomic_write32(&external_data_status_, status);
+  } 
+  return status;
+}
+
+Catalog::ExternalDataStatus Catalog::GetExternalDataUnlocked() const {
+  ExternalDataStatus status = static_cast<ExternalDataStatus>(atomic_read32(&external_data_status_));
+  if (status == Unknown) {
+    pthread_mutex_lock(lock_);
```

Comment:
Currently we sometimes have `pthread_mutex_lock` directly in the code, sometimes we use `MutexLockGuard` from util_concurrency.h.  New code should only use the `MutexLockGuard`.